tests/Admin/Filter/QueryBuilder.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ClassificationBundle\Tests\Admin\Filter;
+
+use Doctrine\ORM\Query\Expr\Andx;
+use Doctrine\ORM\Query\Expr\Orx;
+
+final class QueryBuilder extends \Doctrine\ORM\QueryBuilder
+{
+    public $parameters = [];
+
+    public $query = [];
+
+    public function __construct()
+    {
+    }
+
+    /**
+     * @param string $key
+     * @param mixed  $value
+     */
+    public function setParameter($key, $value, $type = null)
+    {
+        $this->parameters[$key] = $value;
+    }
+
+    public function andWhere()
+    {
+        $query = \func_get_args();
+
+        $this->query[] = (string) $query;
```

Comment:
Isn't this lead to an "array to string conversion" error?

Reply:
It's basically a copy & paste of https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/tests/Filter/QueryBuilder.php

Reply:
I don't see where we are casting the result of `func_get_args()` as string at sonata-project/doctrine-orm-admin-bundle.
Just for the records, here is the example: https://3v4l.org/6UZbV.